src/dfx/src/actors/pocketic.rs

```diff
@@ -300,7 +308,29 @@ fn pocketic_start_thread(
                 }
                 Ok(i) => i,
             };
-            addr.do_send(signals::PocketIcRestarted { port });
+
+            let instance_0 = match initialize_gateway(
+                format!("http://localhost:{port}").parse().unwrap(),
+                config.pocketic_proxy_config.domains.clone(),
+                config.pocketic_proxy_config.bind,
+                logger.clone(),
+            ) {
+                Err(e) => {
+                    error!(logger, "Failed to initialize HTTP gateway: {e:#}");
+                    let _ = child.kill();
+                    let _ = child.wait();
+                    if receiver.try_recv().is_ok() {
+                        debug!(logger, "Got signal to stop.");
+                        break;
+                    } else {
+                        continue;
+                    }
+                }
+                Ok(i) => i,
+            };
+            assert_eq!(instance, instance_0);
```

Comment:
I would drop this assert: the backend instance and gateway instance do not have to have the same instance ID. (If the code relies on the instance ID being zero, then let's assert that condition or fix the code to not assume that.)